// primevue/radiobuttongroup/index2.mjs
import { uuid } from "@primeuix/utils";
import script$2 from "../../_primevue/core/baseeditableholder/index.mjs";
import RadioButtonGroupStyle from "./style/index.mjs";
import { createElementBlock, openBlock, mergeProps, renderSlot } from "vue";
var script$1 = {
  name: "BaseRadioButtonGroup",
  "extends": script$2,
  style: RadioButtonGroupStyle,
  provide: function provide() {
    return {
      $pcRadioButtonGroup: this,
      $parentInstance: this
    };
  }
};
var script = {
  name: "RadioButtonGroup",
  "extends": script$1,
  inheritAttrs: false,
  data: function data() {
    return {
      groupName: this.name
    };
  },
  watch: {
    name: function name(newValue) {
      this.groupName = newValue || uuid("radiobutton-group-");
    }
  },
  mounted: function mounted() {
    this.groupName = this.groupName || uuid("radiobutton-group-");
  }
};
function render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", mergeProps({
    "class": _ctx.cx("root")
  }, _ctx.ptmi("root")), [renderSlot(_ctx.$slots, "default")], 16);
}
script.render = render;
export {
  script as default
};
