// primevue/stepitem/index2.mjs
import script$2 from "../../_primevue/core/basecomponent/index.mjs";
import StepItemStyle from "./style/index.mjs";
import { createElementBlock, openBlock, mergeProps, renderSlot } from "vue";
var script$1 = {
  name: "BaseStepItem",
  "extends": script$2,
  props: {
    value: {
      type: [String, Number],
      "default": void 0
    }
  },
  style: StepItemStyle,
  provide: function provide() {
    return {
      $pcStepItem: this,
      $parentInstance: this
    };
  }
};
var script = {
  name: "StepItem",
  "extends": script$1,
  inheritAttrs: false,
  inject: ["$pcStepper"],
  computed: {
    isActive: function isActive() {
      var _this$$pcStepper;
      return ((_this$$pcStepper = this.$pcStepper) === null || _this$$pcStepper === void 0 ? void 0 : _this$$pcStepper.d_value) === this.value;
    }
  }
};
var _hoisted_1 = ["data-p-active"];
function render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", mergeProps({
    "class": _ctx.cx("root"),
    "data-p-active": $options.isActive
  }, _ctx.ptmi("root")), [renderSlot(_ctx.$slots, "default")], 16, _hoisted_1);
}
script.render = render;
export {
  script as default
};
